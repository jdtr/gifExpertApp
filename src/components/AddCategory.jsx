import { useState } from "react"

export const AddCategory = ({ onNewValue }) => {
 const [inputValue, setInputValue] = useState('');

 const onInputChange = ({target}) => {
    setInputValue(target.value);
    console.log(target.value)
 }

 const onSubmit = (event) => {
     event.preventDefault();
    if ( inputValue.trim().length < 2 ) return;
 
    setInputValue('');
    onNewValue(inputValue.trim())
 }

  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input 
            type="text" 
            placeholder="Search gifs" 
            value={ inputValue }
            onChange={ onInputChange } // Se puede mandar solo la ref de la funciÃ³n
        />
    </form>
  )
}
