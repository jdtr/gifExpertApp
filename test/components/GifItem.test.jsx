import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Test in GifItem', () => {
    const title = 'Saitama';
    const url = "https://one-pounch.com/saitama.jpg";

    test('It should be to do match with snapshop', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect(container).toMatchSnapshot();
    })

    test('It should show title src & alt', () => {
        render(<GifItem title={ title } url={ url } />);
        
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })

    test('It should show title', () => {
        render(<GifItem title={ title } url={ url } />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})