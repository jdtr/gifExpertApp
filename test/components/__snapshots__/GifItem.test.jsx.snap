// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem It should be to do match with snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-pounch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
